### PR TITLE
Ranking thread-safety

### LINKED ISSUE #624: Make reputation system thread safe

### PR DESCRIPTION
**Pull request by paulperegud**
Ranking class is thread-safe as it is. I've added a test and demoted some methods to non-public.

resolves #624

### CODE REVIEW

File: tests/golem/ranking/test_ranking.py [hunk 1]
```diff
@@ -1,8 +1,38 @@
...
+        thread2.start()
+        thread3 = Thread(target=run)
+        thread3.start()
+        thread4 = Thread(target=run)
+        thread4.start()
+        thread1.join()
+        thread2.join()
+        thread3.join()
+        thread4.join()
+        self.assertEqual(0.0, expected)
```
Comment (mfranciszkiewicz) [inline tests/golem/ranking/test_ranking.py]: `expected` is assigned a value prior to the actual test. The outcome doesn't change, though, and the assertion still holds.

Comment (paulperegud) [reply tests/golem/ranking/test_ranking.py]: Fixed. Was trying not to rely on value 0.0 and omitted last step.

File: tests/golem/ranking/test_ranking.py [hunk 2]
```diff
@@ -1,8 +1,39 @@
...
+from threading import Thread
+
 from golem.tools.testwithdatabase import TestWithDatabase
-from golem.ranking.ranking import RankingDatabase
+from golem.ranking.ranking import Ranking, RankingDatabase, RankingStats
+from golem.client import Client
 
 
 class TestRankingDatabase(TestWithDatabase):
+    def test_increase_trust_thread_safety(self):
```
Comment (badb) [inline tests/golem/ranking/test_ranking.py]: This test doesn't work on windows:
1) `OperationalError: database is locked` occurs
2) `AssertionError: 0.01098901098901099 != 0.0`

**Review by badb — CHANGES_REQUESTED** [review]

**Comment by badb** [comment]
I've added some locks so test passes on my machine, but feel free to revert and do something else.

**Pull request merged**